zaza/model.py

```diff
@@ -1130,6 +1170,8 @@ async def async_run_action(unit_name, action_name, model_name=None,
     action_obj = await unit.run_action(action_name, **action_params)
     await action_obj.wait()
     action_obj = _normalise_action_object(action_obj)
+    await async_update_unknown_action_status(model, action_obj)
```

Comment:
I am not sure how this condition arise of status unknown given the L#1171 will wait for status completed or failed [1]. The code in [1] is not changed for 8 years.

If you are seeing this during your test runs, there should be something wrong in python-libjuju action object updates.
+1 for the changes since you are seeing the error in your test runs.

[1] https://github.com/juju/python-libjuju/blob/master/juju/model.py#L1347

Reply:
I imagine it's something to do with the async nature of juju updates from the model to libjuju (any client), as I've seen other transient examples like this.  The `.get_action_status()` does a round-trip to the controller, and so forces the update.